packages/contract/lib/execute.js

```diff
@@ -124,11 +124,12 @@ const execute = {
    * @param  {Array}  inputs     ABI segment defining method arguments
    * @return {Boolean}           true if final argument is `defaultBlock`
    */
-  hasDefaultBlock: function (args, lastArg, inputs) {
+  hasDefaultBlock: function (args, lastArg, methodABI) {
+    const expectedArgs = methodABI ? methodABI.inputs.length : 0;
```

Comment:
A small nit but choose to ignore it; should we name this `expectedNumberArgs` or something? It probably doesn't matter too much but I was confused for a split second.

Reply:
Sure, I can go clarify that.